software/main.py

```diff
@@ -104,198 +136,232 @@ def main_loop():
                 frame = 0
                 end = time.time()
                 fps = 30/(end-start)
-                start = end
-                #[Object: x=7; y=212; size=15.0; distance=212; exists=True]  
+                start = end 
                 print(fps)  
 
+            
+            try:
+            
+                if ref_cmds == True:
+                    run, blue = ref.get_current_referee_command()
+                    print(run, blue )
+                    if run == True and First_Ref==1:
+                        state = State.FIND_BALL
+                        if blue == True:
+                            basket_color = BasketColor.BLUE
+                        else:
+                            basket_color = BasketColor.MAGENTA
+                        First_Ref = 0
+                        continue
+                    elif run == False and First_Ref==0:
+                        state = State.WAITING
+                        First_Ref = 1
+                elif ref_cmds == False and First_Ref==1:
+                    state = State.FIND_BALL
+                    First_Ref = 0
+        
+            except:
+                print("Server client communication failed.")
+            
+
+            if state == State.WAITING:
+                robot.stop()
+                print("WAITIN")
+                
+
+
+
             if state == State.FIND_BALL:
                 
                 print("--------Searching ball--------")
-                if len(processed_Data.balls)!=0:
-                    state=State.MOVE
+                print(processed_Data.balls)
+                if timer:
+                    elapsed_time = time.time() - start_time
+                else:
+                    start_time = time.time()
+                    timer = True
+
+                if elapsed_time > 5 and (processed_Data.basket_m.exists or processed_Data.basket_b.exists):

```

Comment:
This basket_m or basket_b exists check is used in multiple places. Maybe a boolean can be added to the response data to make fetchign it simpler

Reply:
Done